test/e2e/reconciler/partition/partitionset_test.go

```diff
@@ -0,0 +1,332 @@
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(2) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 2 partitions, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partition count to become 2")
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 2 {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 2 partitions, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 2 partitions")
+
+	t.Logf("Excluding the shard of the third region")
+	shard3.Labels = map[string]string{
+		"region":   "partition-test-region-3",
+		"excluded": "true",
+	}
+	_, err = shardClient.Cluster(core.RootCluster.Path()).Update(ctx, shard3, metav1.UpdateOptions{})
+	require.NoError(t, err, "error updating shard")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(1) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partition count to become 1")
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 1 {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 1 partition")
+
+	// The following tests are focused on the admission
+	errorPartitionSet := &topologyv1alpha1.PartitionSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admission-partitionset",
+		},
+		Spec: topologyv1alpha1.PartitionSetSpec{
+			Dimensions: []string{"region"},
+		},
+	}
+
+	t.Logf("Key too long in matchExpressions")
+	errorPartitionSet.Spec.ShardSelector = &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "region/1234567890123456789012345678901234567890123456789012345678901234567890",
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   []string{"antartica", "greenland"},
+			},
+		},
+	}
+	_, err = partitionSetClient.Cluster(partitionClusterPath).Create(ctx, errorPartitionSet, metav1.CreateOptions{})
+	require.Error(t, err, "error creating partitionSet expected")
+
+	t.Logf("Character not allowed at first place in matchExpressions values")
```

Comment:
Not sure if we need to check all incorrect values, i'd rely on k8s's validation.
From our perspective the most important thing is to make sure an invalid request will be rejected and won't have any side effect (i.e. deleting a partition).